sprokit/processes/core/video_input_process.cxx

```diff
@@ -151,20 +172,48 @@ ::_init()
 }
 
 
-// ----------------------------------------------------------------
+// -----------------------------------------------------------------------------
 void video_input_process
 ::_step()
 {
   kwiver::vital::timestamp ts;
 
-  if ( d->m_video_reader->next_frame( ts ) )
+  bool frame_read = false;
+  bool bad_frame = false;
+
+  try
+  {
+    frame_read = d->m_video_reader->next_frame( ts );
+  }
+  catch( const kwiver::vital::image_exception& e )
+  {
+    if ( d->m_exit_on_invalid || d->m_first_frame )
+    {
+      throw e;
+    }
+    else
+    {
+      frame_read = true;
+      bad_frame = true;
+
+      LOG_WARN( logger(), "Video source skipping corrupt frame." );
```

Comment:
It may be useful for debugging to include which frame was skipped, it that information is available in a useful form.